src/tsolvers/egraph/EnodeStore.cc

```diff
@@ -109,6 +120,88 @@ ERef EnodeStore::addList(ERef x, ERef y) {
     }
     return rval;
 }
+/**
+ * Construct an Enode for a given PTRef, assuming that all the child PTRefs have
+ * already been introduced an Enode.  In case of a Boolean return valued Enode,
+ * add also an enode of the negation of the PTRef.  If the Boolean Enode is
+ * non-atomic, no child Enodes will be constructed.
+ * @param tr The PTRef for which the Enode(s) will be constructed
+ * @return a vector of <PTRef,ERef> pairs consisting either of a single element
+ * if the PTRef is non-boolean; two elements, the first of which corresponds to
+ * the positive form and and the second the negated form of the PTRef tr; or is empty
+ * if the PTRef has already been inserted.
+ */
+vec<std::pair<PTRef,ERef>> EnodeStore::constructTerm(PTRef tr) {
+
+    if (termToERef.has(tr))
+        return {};
```

Comment:
I have some conflicting feelings about this. I see the benefit of EnodeStore enforcing invariant directly, instead of relying on users of EnodeStore to ensure the invariant.

However, since the user of EnodeStore is basically just the EGraph, I think it would not be a bad decision guard there that EnodeStore is not asked to process the same term twice.

Reply:
As discussed, we keep this so that EnodeStore enforces its own invariant.